emacs/config.org
#+TITLE: GNU EMACS config
#+AUTHOR: Maximiliano Christener
#+STARTUP: overview
#+OPTIONS: toc:2

* Acerca de

Esta es mi configuración de GNU Emacs.

** Instalación
Una vez hecha la instación de GNU Emacs (vanilla), se instalan
dependencias necesarias: =cmake=, =libvterm=, =hunspell= y =hunspell-es_ar=.
Tener también en cuenta la fuente.

Luego, crear las carpetas =autosaves= y =backups= en =~/.config/emacs/=.
También guardar una copia de este =config.org=. Para que todo funcione,
en nuestro archivo =init.el= debemos escribir

#+begin_example
(org-babel-load-file
  (expand-file-name
    "config.org"
    user-emacs-directory))
#+end_example

Ya podemos iniciar Emacs por primera vez. Tras hacerlo, correr =M-x
all-the-icons-install-fonts= y, en Bash, =fc-cache -f -v=.

** Algunas características

- =C-c C-r m= activa el modo oscuro en PDFs

** Sobre Org Mode

Los modos son
- =C-c C-x a= para archivar
- =C-c TAB= para abrir el árbol
- =C-c o= para ir a un lugar determinado
  desde la toc.

*** Expansions in Org-Tempo

Org-tempo agrega un snippet para escribir bloques en Org. A saber,

| Typing the below + TAB | Expands to ...                          |
|------------------------+-----------------------------------------|
| <a                     | '#+BEGIN_EXPORT ascii' … '#+END_EXPORT  |
| <c                     | '#+BEGIN_CENTER' … '#+END_CENTER'       |
| <C                     | '#+BEGIN_COMMENT' … '#+END_COMMENT'     |
| <e                     | '#+BEGIN_EXAMPLE' … '#+END_EXAMPLE'     |
| <E                     | '#+BEGIN_EXPORT' … '#+END_EXPORT'       |
| <h                     | '#+BEGIN_EXPORT html' … '#+END_EXPORT'  |
| <l                     | '#+BEGIN_EXPORT latex' … '#+END_EXPORT' |
| <q                     | '#+BEGIN_QUOTE' … '#+END_QUOTE'         |
| <s                     | '#+BEGIN_SRC' … '#+END_SRC'             |
| <v                     | '#+BEGIN_VERSE' … '#+END_VERSE'         |




* Código
#+begin_src emacs-lisp

     ;; MELPA
     (require 'package)
     (add-to-list 'package-archives '("melpa" . "https://melpa.org/packages/") t)
     (add-to-list 'package-archives '("melpa-stable" . "https://stable.melpa.org/packages/") t)
     (package-initialize)
     (setq package-pinned-packages '((all-the-icons . "melpa-stable")
				       (all-the-icons-dired . "melpa-stable")
				       (all-the-icons-completion . "melpa-stable")))

     ;; evil mode
     (use-package evil
       :ensure t)
     (require 'evil)
     (evil-mode 1)
     (setq evil-default-state 'emacs)
     (define-key evil-insert-state-map (kbd "M-j") #'evil-normal-state)


     ;; Font Settings
     (set-face-attribute 'default nil :font "UbuntuMono Nerd Font" :height 160 :weight 'medium)
     (set-face-attribute 'variable-pitch nil :font "DejaVu Sans" :height 160 :weight 'medium)
     (set-face-attribute 'fixed-pitch nil :font "UbuntuMono Nerd Font" :height 160 :weight 'medium)
     (setq-default line-spacing 0.12)
     (set-face-attribute 'font-lock-comment-face nil :slant 'italic) ;; comentarios en itálica

     ;; all-the-icons
     (use-package all-the-icons
     :ensure t
     :if (display-graphic-p))
     (use-package all-the-icons-completion :ensure t :defer t)
     (all-the-icons-completion-mode)


     ;; Keybindings, e.g, (keymap-set c-mode-map "C-S-l" 'function-name)
     (keymap-global-set "C-x C-a" 'auto-fill-mode)
     (setq parens-require-spaces nil) ;; para quitar los espacios de M-(

     ;; company-mode
     (use-package company :ensure t
      :config (add-hook 'after-init-hook 'global-company-mode))

     ;; eglot
     (add-hook 'ess-mode-hook 'eglot-ensure)
     (add-hook 'tex-mode-hook 'eglot-ensure)  
     (add-hook 'bibtex-mode-hook 'eglot-ensure)

     ;; spell
     (setq ispell-program-name "hunspell")
     (setq ispell-dictionary "es_AR")
     ; ido-mode
;     (ido-mode t)

     ;; Appearance
     (setq inhibit-startup-screen t)
     (tool-bar-mode -1)
     (scroll-bar-mode -1)
     (menu-bar-mode -1)
     (blink-cursor-mode -1)
     (delete-selection-mode +1)
     (global-goto-address-mode +1)
     (set-face-attribute 'default nil :height 138) ; default zoom
     (add-to-list 'default-frame-alist '(alpha-background . 92)) ; transparency
     (setq case-fold-search t) ; case-insensitive search
     (setq sentence-end-double-space nil)
     ;; Theme
     (setq modus-themes-paren-match '(bold intense)
	   modus-themes-scale-headings t
	   modus-themes-headings
	   '((1 . (rainbow bold))
	     (2 . (rainbow bold))))
     ;;      (3 . (rainbow bold 1.2))
     ;;      (t . (semilight 1.1)))

     (load-theme 'modus-vivendi t)
     ;; number styling
     (setq display-line-numbers-type 'relative)
     (global-display-line-numbers-mode 1)  

     ;; AUCTeX
     ; (setq-default TeX-master nil)
     (setq TeX-auto-save t)
     (setq TeX-parse-self t)
     (add-hook 'LaTeX-mode-hook 'auto-fill-mode)
     ; reftex
     (add-hook 'LaTeX-mode-hook 'turn-on-reftex) ; reftex
       (setq reftex-plug-into-AUCTeX t)
      ; zathura and synctex
     (require 'tex)
     (add-hook 'LaTeX-mode-hook 'TeX-source-correlate-mode)
     (setq TeX-PDF-mode t
	   TeX-source-correlate-mode t
	   TeX-source-correlate-start-server t)
     (add-to-list 'TeX-expand-list
		  '("%sn" (lambda () server-name)))
     (add-to-list 'TeX-view-program-list
		  '("Zathura"
		    ("zathura %o"
		     (mode-io-correlate " --synctex-forward %n:0:\"%b\" -x \"emacsclient --socket-name=%sn +%{line} %{input}\""))
		    "zathura"))
     (setcar (cdr (assoc 'output-pdf TeX-view-program-selection)) "Zathura")



     ;; ESS
     (use-package ess :ensure t)
     (org-babel-do-load-languages 'org-babel-load-languages '((R . t)))
     (setq ess-swv-pdflatex-commands
	   '((output-pdf "pdflatex -interaction nonstopmode -encoding utf8 %S%(PDFout)")))
     (setq ess-swv-pdflatex-options
	   '("-shell-escape" "-file-line-error" "-synctex=1"))
     (setq ess-swv-weave-options
	   '(("noweb" "yes" "encoding" "utf8")))
     (show-paren-mode)
     ; keybinds Rdired buffer with F9:
     (add-hook 'ess-r-mode-hook
	     '(lambda ()
		(local-set-key (kbd "<f9>") #'ess-rdired)))
     (add-hook 'ess-rdired-mode-hook
	     '(lambda ()
		(local-set-key (kbd "<f9>") #'kill-buffer-and-window)))
     ; pane configs
     (setq display-buffer-alist
	 '(("*R Dired"
	    (display-buffer-reuse-window display-buffer-at-bottom)
	    (window-width . 0.5)
	    (window-height . 0.25)
	    (reusable-frames . nil))
	   ("*R"
	    (display-buffer-reuse-window display-buffer-in-side-window)
	    (side . right)
	    (slot . -1)
	    (window-width . 0.5)
	    (reusable-frames . nil))
	   ("*Help"
	    (display-buffer-reuse-window display-buffer-in-side-window)
	    (side . right)
	    (slot . 1)
	    (window-width . 0.5)
	    (reusable-frames . nil))) )


     ;; Python
     (add-hook 'python-mode-hook
	       (lambda ()
		 (define-key python-mode-map (kbd "C-<return>") 'python-shell-send-statement)
		 (define-key python-mode-map (kbd "C-c C-e") nil)))



     ;; Enable Auto-save and Backup Files in Specified Directory
     (setq auto-save-default t) ; Enable auto-save
     (setq make-backup-files t) ; Enable backup files
     (setq auto-save-file-name-transforms
	   `((".*" ,(expand-file-name "~/.config/emacs/autosaves/") t)))
     (setq backup-directory-alist
	   `(("." . ,(expand-file-name "~/.config/emacs/backups/"))))





     ;;
     ;; Org-mode
					     ; Appearance
     (setq org-format-latex-options (plist-put org-format-latex-options :scale 2.4))
     (setq org-hide-emphasis-markers t)
     (custom-theme-set-faces
      'user
      '(variable-pitch ((t (:family "UbuntuMono Nerd Font" :height 180))))
      '(fixed-pitch ((t (:family "UbuntuMono Nerd Font" :height 160)))))
     (add-hook 'org-mode-hook 'variable-pitch-mode)
     (setq org-emphasis-alist
	   '(("*" (bold :foreground "Orange"))
	     ("/" (italic :foreground "Green"))
	     ("_" underline)
	     ("=" (:background "maroon" :foreground "white"))
	     ("~" (:background "deep sky blue" :foreground "MidnightBlue"))
	     ("+" (:strike-through t))))
     (setq org-hide-emphasis-markers t)
     (dolist (face '((org-level-1 . 1.2)
		     (org-level-2 . 1.1)
		     (org-level-3 . 1.05)
		     (org-level-4 . 1.0)
		     (org-level-5 . 1.1)
		     (org-level-6 . 1.1)
		     (org-level-7 . 1.1)
		     (org-level-8 . 1.1)))
       (set-face-attribute (car face) nil
			   :font "UbuntuMono Nerd Font"
			   :weight 'medium
			   :height (cdr face)))
     (setq header-line-format nil)
					     ; Presentation in Org-mode
     (use-package org-tree-slide :ensure t :custom (org-image-actual-width nil))
					     ; TOC
     (use-package toc-org :ensure t :commands toc-org-enable)
     (add-hook 'org-mode-hook 'toc-org-enable)
     (add-hook 'org-mode-hook 'org-indent-mode)
					     ; Bullets
     (use-package org-bullets :ensure t)
     (add-hook 'org-mode-hook (lambda () (org-bullets-mode 1)))
					     ; Presentations
     (use-package org-tree-slide
       :ensure t
       :custom (org-image-actual-width nil))
     (setq header-line-format " ")
					     ; en html
     (use-package ox-reveal
       :ensure t)
      ;
     (require 'org-tempo)
     ;;

     ;; vterm
     (use-package vterm
       :ensure t)


     ;; Dired
     (use-package all-the-icons-dired
             :ensure t)
     (add-hook 'dired-mode-hook 'all-the-icons-dired-mode)
     (setq dired-listing-switches "-lt --dired --group-directories-first -h -G") ; lt es ordenado por tiempo
     (setq dired-guess-shell-alist-user
      '(("\\.\\(png\\|jpe?g\\|tiff\\)" "sxiv" "xdg-open") ; abrir con ! o &
        ("\\.\\(mp[34]\\|m4a\\|ogg\\|flac\\|webm\\|mkv\\)" "mpv" "xdg-open")
		(".*" "xdg-open")))
     (setq dired-dwim-target t) ; para copiar en la otra ventana
     (add-hook 'dired-mode-hook 'dired-hide-details-mode)
     (add-hook 'dired-mode-hook (lambda () (dired-omit-mode)))
     ; 
     (file-name-shadow-mode 1)


     ;; PDF Tools
     (use-package pdf-tools :ensure t :defer t)
     (pdf-tools-install)
     (add-hook 'pdf-view-mode-hook (lambda () (display-line-numbers-mode -1)))



#+end_src



config/emacs/config.org
#+TITLE: GNU EMACS config
#+AUTHOR: Maximiliano Christener
#+STARTUP: overview
#+OPTIONS: toc:2

* Acerca de

Esta es mi configuración de GNU Emacs.

** Instalación
Una vez hecha la instación de GNU Emacs (vanilla), se instalan
dependencias necesarias: =cmake=, =libvterm=, =hunspell= y =hunspell-es_ar=.
Tener también en cuenta la fuente.

Luego, crear las carpetas =autosaves= y =backups= en =~/.config/emacs/=.
También guardar una copia de este =config.org=. Para que todo funcione,
en nuestro archivo =init.el= debemos escribir

#+begin_example
(org-babel-load-file
  (expand-file-name
    "config.org"
    user-emacs-directory))
#+end_example

Ya podemos iniciar Emacs por primera vez. Tras hacerlo, correr =M-x
all-the-icons-install-fonts= y, en Bash, =fc-cache -f -v=.

** Algunas características

- =C-c C-r m= activa el modo oscuro en PDFs

** Sobre Org Mode

Los modos son
- =C-c C-x a= para archivar
- =C-c TAB= para abrir el árbol
- =C-c o= para ir a un lugar determinado
  desde la toc.

*** Expansions in Org-Tempo

Org-tempo agrega un snippet para escribir bloques en Org. A saber,

| Typing the below + TAB | Expands to ...                          |
|------------------------+-----------------------------------------|
| <a                     | '#+BEGIN_EXPORT ascii' … '#+END_EXPORT  |
| <c                     | '#+BEGIN_CENTER' … '#+END_CENTER'       |
| <C                     | '#+BEGIN_COMMENT' … '#+END_COMMENT'     |
| <e                     | '#+BEGIN_EXAMPLE' … '#+END_EXAMPLE'     |
| <E                     | '#+BEGIN_EXPORT' … '#+END_EXPORT'       |
| <h                     | '#+BEGIN_EXPORT html' … '#+END_EXPORT'  |
| <l                     | '#+BEGIN_EXPORT latex' … '#+END_EXPORT' |
| <q                     | '#+BEGIN_QUOTE' … '#+END_QUOTE'         |
| <s                     | '#+BEGIN_SRC' … '#+END_SRC'             |
| <v                     | '#+BEGIN_VERSE' … '#+END_VERSE'         |




* Código
#+begin_src emacs-lisp

    ;; MELPA
    (require 'package)
    (add-to-list 'package-archives '("melpa" . "https://melpa.org/packages/") t)
    (add-to-list 'package-archives '("melpa-stable" . "https://stable.melpa.org/packages/") t)
    (package-initialize)
    (setq package-pinned-packages '((all-the-icons . "melpa-stable")
                                      (all-the-icons-dired . "melpa-stable")
                                      (all-the-icons-completion . "melpa-stable")))


    ;; Font Settings
    (set-face-attribute 'default nil :font "UbuntuMono Nerd Font" :height 160 :weight 'medium)
    (set-face-attribute 'variable-pitch nil :font "DejaVu Sans" :height 160 :weight 'medium)
    (set-face-attribute 'fixed-pitch nil :font "UbuntuMono Nerd Font" :height 160 :weight 'medium)
    (setq-default line-spacing 0.12)
    (set-face-attribute 'font-lock-comment-face nil :slant 'italic) ;; comentarios en itálica


    ;; all-the-icons
    (use-package all-the-icons
    :ensure t
    :if (display-graphic-p))
    (use-package all-the-icons-completion :ensure t :defer t)
    (all-the-icons-completion-mode)


    ;; Dired
    (with-eval-after-load 'dired
      (require 'dired-x)
    ; Set dired-x global variables here. For example: (setq dired-x-hands-off-my-keys nil)
    )
    (setq dired-listing-switches
          "-lt --dired --group-directories-first -h -G") ; lt es ordenado por tiempo
    (setq dired-dwim-target t) ; para copiar en la otra ventana
    (add-hook 'dired-mode-hook
              'dired-hide-details-mode (lambda () (dired-omit-mode)))
    (file-name-shadow-mode 1)
    ;
    (use-package all-the-icons-dired
          :ensure t)
    (add-hook 'dired-mode-hook 'all-the-icons-dired-mode)
    ;
    (setq dired-guess-shell-alist-user
     '(("\\.\\(png\\|jpe?g\\|tiff\\)" "sxiv" "xdg-open") ; abrir con ! o &
       ("\\.\\(mp[34]\\|m4a\\|ogg\\|flac\\|webm\\|mkv\\)" "mpv" "xdg-open")
               (".*" "xdg-open")))


    ;; Keybindings, e.g, (keymap-set c-mode-map "C-S-l" 'function-name)
    (keymap-global-set "C-x C-a" #'auto-fill-mode)
    (setq parens-require-spaces nil) ;; para quitar los espacios de M-(

    ;; company-mode
    (use-package company :ensure t
     :config (add-hook 'after-init-hook 'global-company-mode))

    ;; eglot
    (add-hook 'ess-mode-hook 'eglot-ensure)
    (add-hook 'tex-mode-hook 'eglot-ensure)  
    (add-hook 'bibtex-mode-hook 'eglot-ensure)

    ;; spell
    (setq ispell-program-name "hunspell")
    (setq ispell-dictionary "es_AR")

    ;; Appearance & misc
    (setq inhibit-startup-screen t)
    (tool-bar-mode -1)
    (scroll-bar-mode -1)
    (menu-bar-mode -1)
    (blink-cursor-mode -1)
    (delete-selection-mode +1)
    (global-goto-address-mode +1)
    (set-face-attribute 'default nil :height 138) ; default zoom
    (add-to-list 'default-frame-alist '(alpha-background . 92)) ; transparency
    (setq case-fold-search t) ; case-insensitive search
    (setq sentence-end-double-space nil)
    ;; Theme
    (setq modus-themes-paren-match '(bold intense)
          modus-themes-scale-headings t
          modus-themes-headings
          '((1 . (rainbow bold))
            (2 . (rainbow bold))))
    ;;      (3 . (rainbow bold 1.2))
    ;;      (t . (semilight 1.1)))
    ;; theme
    (load-theme 'modus-vivendi t)
    ;; number styling
    (setq display-line-numbers-type 'relative)
    (global-display-line-numbers-mode 1)  
    ;; Enable Auto-save and Backup Files in Specified Directory
    (setq auto-save-default t) ; Enable auto-save
    (setq make-backup-files t) ; Enable backup files
    (setq auto-save-file-name-transforms
          `((".*" ,(expand-file-name "~/.config/emacs/autosaves/") t)))
    (setq backup-directory-alist
          `(("." . ,(expand-file-name "~/.config/emacs/backups/"))))


    ;; AUCTeX
    (setq-default TeX-master nil)
    (setq TeX-auto-save t)
    (setq TeX-parse-self t)
    (add-hook 'LaTeX-mode-hook 'auto-fill-mode)
    ; reftex
    (add-hook 'LaTeX-mode-hook 'turn-on-reftex) ; reftex
      (setq reftex-plug-into-AUCTeX t)
     ; zathura and synctex
    (require 'tex)
    (add-hook 'LaTeX-mode-hook 'TeX-source-correlate-mode)
    (setq TeX-PDF-mode t
          TeX-source-correlate-mode t
          TeX-source-correlate-start-server t)
    (add-to-list 'TeX-expand-list
                 '("%sn" (lambda () server-name)))
    (add-to-list 'TeX-view-program-list
                 '("Zathura"
                   ("zathura %o"
                    (mode-io-correlate " --synctex-forward %n:0:\"%b\" -x \"emacsclient --socket-name=%sn +%{line} %{input}\""))
                   "zathura"))
    (setcar (cdr (assoc 'output-pdf TeX-view-program-selection)) "Zathura")



    ;; ESS
    (use-package ess :ensure t)
    (org-babel-do-load-languages 'org-babel-load-languages '((R . t)))
    (setq ess-swv-pdflatex-commands
          '((output-pdf "pdflatex -interaction nonstopmode -encoding utf8 %S%(PDFout)")))
    (setq ess-swv-pdflatex-options
          '("-shell-escape" "-file-line-error" "-synctex=1"))
    (setq ess-swv-weave-options
          '(("noweb" "yes" "encoding" "utf8")))
    (show-paren-mode)
    ; keybinds Rdired buffer with F9:
    (add-hook 'ess-r-mode-hook
            (lambda ()
               (local-set-key (kbd "<f9>") #'ess-rdired)))
    (add-hook 'ess-rdired-mode-hook
            (lambda ()
               (local-set-key (kbd "<f9>") #'kill-buffer-and-window)))
    ; pane configs
    (setq display-buffer-alist
        '(("*R Dired"
           (display-buffer-reuse-window display-buffer-at-bottom)
           (window-width . 0.5)
           (window-height . 0.25)
           (reusable-frames . nil))
          ("*R"
           (display-buffer-reuse-window display-buffer-in-side-window)
           (side . right)
           (slot . -1)
           (window-width . 0.5)
           (reusable-frames . nil))
          ("*Help"
           (display-buffer-reuse-window display-buffer-in-side-window)
           (side . right)
           (slot . 1)
           (window-width . 0.5)
           (reusable-frames . nil))) )


    ;; Python
    (add-hook 'python-mode-hook
              (lambda ()
                (define-key python-mode-map (kbd "C-<return>") #'python-shell-send-statement)
                (define-key python-mode-map (kbd "C-c C-e") nil)))





    ;;
    ;; Org-mode
                                            ; Appearance
    (setq org-format-latex-options (plist-put org-format-latex-options :scale 2.4))
    (setq org-hide-emphasis-markers t)
    (custom-theme-set-faces
     'user
     '(variable-pitch ((t (:family "UbuntuMono Nerd Font" :height 180))))
     '(fixed-pitch ((t (:family "UbuntuMono Nerd Font" :height 160)))))
    (add-hook 'org-mode-hook 'variable-pitch-mode)
    (setq org-emphasis-alist
          '(("*" (bold :foreground "Orange"))
            ("/" (italic :foreground "Green"))
            ("_" underline)
            ("=" (:background "maroon" :foreground "white"))
            ("~" (:background "deep sky blue" :foreground "MidnightBlue"))
            ("+" (:strike-through t))))
    (setq org-hide-emphasis-markers t)
    (dolist (face '((org-level-1 . 1.2)
                    (org-level-2 . 1.1)
                    (org-level-3 . 1.05)
                    (org-level-4 . 1.0)
                    (org-level-5 . 1.1)
                    (org-level-6 . 1.1)
                    (org-level-7 . 1.1)
                    (org-level-8 . 1.1)))
      (set-face-attribute (car face) nil
                          :font "UbuntuMono Nerd Font"
                          :weight 'medium
                          :height (cdr face)))
    (setq header-line-format nil)
                                            ; Presentation in Org-mode
    (use-package org-tree-slide :ensure t :custom (org-image-actual-width nil))
                                            ; TOC
    (use-package toc-org :ensure t :commands toc-org-enable)
    (add-hook 'org-mode-hook 'toc-org-enable)
    (add-hook 'org-mode-hook 'org-indent-mode)
                                            ; Bullets
    (use-package org-bullets :ensure t)
    (add-hook 'org-mode-hook (lambda () (org-bullets-mode 1)))
                                            ; Presentations
    (use-package org-tree-slide
      :ensure t
      :custom (org-image-actual-width nil))
    (setq header-line-format " ")
                                            ; en html
    (use-package ox-reveal
      :ensure t)
     ;
    (require 'org-tempo)
    ;;



    ;; vterm
    (use-package vterm
      :ensure t)


    ;; evil mode
    (use-package evil
      :ensure t)
    (require 'evil)
    (evil-mode 1)
    (setq evil-default-state 'emacs)
    (define-key evil-insert-state-map (kbd "M-j") #'evil-normal-state)


    ;; PDF Tools
    (use-package pdf-tools :ensure t :defer t)
    (pdf-tools-install)
    (add-hook 'pdf-view-mode-hook (lambda () (display-line-numbers-mode -1)))



#+end_src



emacs/.config/emacs/config.el
(require 'package)
  (add-to-list 'package-archives '("melpa" . "https://melpa.org/packages/") t)
  (package-initialize)
;; (setq package-check-signature nil) ;; chequea las firmas digitales

; basic mode
  (tool-bar-mode -1 )
  (scroll-bar-mode -1)
  (menu-bar-mode -1)
  (blink-cursor-mode -1)
  (delete-selection-mode +1)
  (global-goto-address-mode +1) ;; allows to access links

  ; Set default zoom
  (set-face-attribute 'default nil :height 138)

  ; initial frame defaults
;;  (setq initial-frame-alist '((top . 1)
;;			    (left . 450)
;;			    (width . 101)
;;			    (height . 70)))
;;

  ; case-insensitive search
  (setq case-fold-search t)

  ; emacs cree que las oraciones terminan con un
  ; punto y dos espacios. Le decimos que no es así.
  (setq sentence-end-double-space nil)

(load-theme 'modus-vivendi t)

(set-face-attribute 'default nil
      :font "JetBrains Mono Nerd Font"
      :height 160
      :weight 'medium)
    (set-face-attribute 'variable-pitch nil
      :font "DejaVu Sans"
      :height 160
      :weight 'medium)
    (set-face-attribute 'fixed-pitch nil
      :font "JetBrains Mono Nerd Font"
      :height 160
      :weight 'medium)
    ;; Makes commented text and keywords italics.
    ;; This is working in emacsclient but not emacs.
    ;; Your font must have an italic face available.
    (set-face-attribute 'font-lock-comment-face nil
      :slant 'italic)
    (set-face-attribute 'font-lock-keyword-face nil
      :slant 'italic)

    ;; This sets the default font on all graphical frames created after restarting Emacs.
    ;; Does the same thing as 'set-face-attribute default' above, but emacsclient fonts
    ;; are not right unless I also add this method of setting the default font.
;    (add-to-list 'default-frame-alist '(font . "JetBrains Mono-16"))

    ;; Uncomment the following line if line spacing needs adjusting.
    (setq-default line-spacing 0.12)

;; auto-fill mode kbd
(global-set-key (kbd "C-x C-a") 'auto-fill-mode)

; ranger
(global-set-key (kbd "C-x r a") 'ranger)

(setq dired-listing-switches "-alt --dired --group-directories-first -h -G")

;; icons
  (use-package all-the-icons
  :ensure t
  :if (display-graphic-p))

(use-package all-the-icons-dired
  :ensure t
  :hook (dired-mode . (lambda () (all-the-icons-dired-mode t))))

;; hide details
(add-hook 'dired-mode-hook 'dired-hide-details-mode)

(use-package ranger
    :ensure t)

  ;; configs
  (setq ranger-cleanup-on-disable t)

  ;; previews
  (setq ranger-preview-file t)
;;  (setq ranger-show-literal t)

(use-package transpose-frame
  :ensure t)

(use-package auctex
  :defer t
  :ensure t)

(setq TeX-auto-save t)
(setq TeX-parse-self t)
(setq-default TeX-master nil)

(use-package auctex-latexmk
  :defer t
  :ensure t)

(require 'auctex-latexmk)
  (auctex-latexmk-setup)

(use-package ess
  :ensure t
  :init (require 'ess-site))

(use-package pdf-tools
  :ensure t
  :defer t)

(pdf-tools-install)

(use-package toc-org
    :ensure t
    :commands toc-org-enable
    :init (add-hook 'org-mode-hook 'toc-org-enable))

(add-hook 'org-mode-hook 'org-indent-mode)
(use-package org-bullets
  :ensure t)

(add-hook 'org-mode-hook (lambda () (org-bullets-mode 1)))

(require 'org-tempo)

(setq org-format-latex-options (plist-put org-format-latex-options :scale 2.4))
